libsignal-protocol/src/pre_key_bundle.rs

```diff
@@ -177,18 +177,16 @@ impl PreKeyBundleBuilder {
         }
     }
 
-    fn get_pre_key(&self) -> Result<(u32, *mut sys::ec_public_key), Error> {
+    fn get_pre_key(&self) -> (u32, *mut sys::ec_public_key) {
         if let PreKeyBundleBuilder {
```

Comment:
Oooh, what that pattern called ? A destruction ! Like in js ? WOW, i think i hadn't a chance to try it before in Rust 😄

Reply:
It's just an application of Rust's normal pattern matching. JS often refer to this sort of thing as *Destructuring*.